pkg/threescale/backend/backend.go

```diff
@@ -430,7 +447,7 @@ func (b *Backend) reportGroupedApps(service api.Service, apps []*Application) []
 	}
 	_, err := b.remoteReport(req)
 	if err != nil {
-		// TODO Log error
+		b.logger.Printf("report failed for service %s and backend %s", string(req.Service), b.client.GetPeer())
```

Comment:
The problem with this logger as defined is that all messages are logged with the same level, which means we are forced to use it both for notification of normal events and error/critical/fatal events, ie. you'd notify that a new backend is being cached, which would be an expected event, just the same as 3scale being unreachable, which would be a (transient?) error, and possibly the same as `CPU0 on fire`, which is likely to be a massive issue.

I think it is important that the caller can specify how critical the condition triggering the message is. To make it more ergonomic you could just define a few more functions in the interface named by level. What do you think?

Reply:
yeah, I had definitely thought about that and see your point. I was just trying to keep it simple to be honest but I think there is no harm to expose maybe `info` and `error` at the minute, what do you think?

Reply:
Agreed. I would also add `debug`, and I think that along with `info` and `error` are likely to cover all needs, but up to you.

Reply:
yes sounds good, ill add them and it fits the current zap logger easily also for our purposes.